### PR TITLE
Add NULL to readme

### PR DESCRIPTION
**Pull request by rakshith91**

### CODE REVIEW

File: sdk/core/azure-core/README.md [hunk 1]
```diff
@@ -179,6 +179,17 @@ class MyCustomEnum(with_metaclass(CaseInsensitiveEnumMeta, str, Enum)):
...
 
+#### Null Sentinel Value
+
+A falsy sentinel object which is supposed to be used to specify attributes
+with no data. This gets serialized to `null` on the wire.
+
+```python
+from azure.core.serialization import NULL
+
+assert bool(NULL) is False
```
Comment (rakshith91) [inline sdk/core/azure-core/README.md:190]:
```suggestion
from azure.core.serialization import NULL

assert bool(NULL) is False

model = Model(
    attr=NULL
)
```

File: sdk/core/azure-core/README.md [hunk 2]
```diff
@@ -179,6 +179,21 @@ class MyCustomEnum(with_metaclass(CaseInsensitiveEnumMeta, str, Enum)):
     BAR = 'bar'
 ```
 
+#### Null Sentinel Value
```
Comment (xiangyan99) [inline sdk/core/azure-core/README.md:182]: Just want to double check we expect this to be used by end user or developer or both?

Comment (rakshith91) [reply sdk/core/azure-core/README.md:182]: we expect this to be used by end user - if they want to pass an explicit null on the wire

Comment (rakshith91) [reply sdk/core/azure-core/README.md:182]: it's a rare edge case (which cloud events happen to allow), so decided to add a sentinel

**Review by xiangyan99 — APPROVED** [review]

**Pull request merged**